### ISSUE TITLE
Site crashes after clicking on "Plugin Keys" , what could be the issue?

### ISSUE THREAD

**Issue by rhlkmth** [mckaywrigley/chatbot-ui#537]

**Comment by rhlkmth** [comment]
### Application error: a client-side exception has occurred (see the browser console for more  information)

**Comment by Cornoualis** [comment]
Same for me, here's the console logs:

```
TypeError: n.find is not a function
    at eH (index-d8d60512342f9171.js:1:47978)
    at ab (framework-73b8966a3c579ab0.js:9:60917)
    at ud (framework-73b8966a3c579ab0.js:9:72803)
    at i (framework-73b8966a3c579ab0.js:9:120372)
    at oO (framework-73b8966a3c579ab0.js:9:99114)
    at framework-73b8966a3c579ab0.js:9:98981
    at oF (framework-73b8966a3c579ab0.js:9:98988)
    at ox (framework-73b8966a3c579ab0.js:9:95740)
    at oC (framework-73b8966a3c579ab0.js:9:96129)
    at r4 (framework-73b8966a3c579ab0.js:9:44778)
a5 @ framework-73b8966a3c579ab0.js:9
a.componentDidCatch.t.callback @ framework-73b8966a3c579ab0.js:9
lH @ framework-73b8966a3c579ab0.js:9
uY @ framework-73b8966a3c579ab0.js:9
e @ framework-73b8966a3c579ab0.js:9
(anonymous) @ framework-73b8966a3c579ab0.js:9
oI @ framework-73b8966a3c579ab0.js:9
oC @ framework-73b8966a3c579ab0.js:9
r4 @ framework-73b8966a3c579ab0.js:9
(anonymous) @ framework-73b8966a3c579ab0.js:9
main-f6414acab6dce0b8.js:1 
TypeError: n.find is not a function
    at eH (index-d8d60512342f9171.js:1:47978)
    at ab (framework-73b8966a3c579ab0.js:9:60917)
    at ud (framework-73b8966a3c579ab0.js:9:72803)
    at i (framework-73b8966a3c579ab0.js:9:120372)
    at oO (framework-73b8966a3c579ab0.js:9:99114)
    at framework-73b8966a3c579ab0.js:9:98981
    at oF (framework-73b8966a3c579ab0.js:9:98988)
    at ox (framework-73b8966a3c579ab0.js:9:95740)
    at oC (framework-73b8966a3c579ab0.js:9:96129)
    at r4 (framework-73b8966a3c579ab0.js:9:44778)
K @ main-f6414acab6dce0b8.js:1
fn @ main-f6414acab6dce0b8.js:1
componentDidCatch @ main-f6414acab6dce0b8.js:1
a.componentDidCatch.t.callback @ framework-73b8966a3c579ab0.js:9
lH @ framework-73b8966a3c579ab0.js:9
uY @ framework-73b8966a3c579ab0.js:9
e @ framework-73b8966a3c579ab0.js:9
(anonymous) @ framework-73b8966a3c579ab0.js:9
oI @ framework-73b8966a3c579ab0.js:9
oC @ framework-73b8966a3c579ab0.js:9
r4 @ framework-73b8966a3c579ab0.js:9
(anonymous) @ framework-73b8966a3c579ab0.js:9
main-f6414acab6dce0b8.js:1 
A client-side exception has occurred, see here for more info: https://nextjs.org/docs/messages/client-side-exception-occurred
K @ main-f6414acab6dce0b8.js:1
fn @ main-f6414acab6dce0b8.js:1
componentDidCatch @ main-f6414acab6dce0b8.js:1
a.componentDidCatch.t.callback @ framework-73b8966a3c579ab0.js:9
lH @ framework-73b8966a3c579ab0.js:9
uY @ framework-73b8966a3c579ab0.js:9
e @ framework-73b8966a3c579ab0.js:9
(anonymous) @ framework-73b8966a3c579ab0.js:9
oI @ framework-73b8966a3c579ab0.js:9
oC @ framework-73b8966a3c579ab0.js:9
r4 @ framework-73b8966a3c579ab0.js:9
(anonymous) @ framework-73b8966a3c579ab0.js:9
```

**Comment by Designdocs** [comment]
Reference: #502

**Comment by gaduffl** [comment]
Same here, I get this error:

Unhandled Runtime Error
TypeError: pluginKeys.find is not a function

Source
components\Chatbar\components\PluginKeys.tsx (95:25) @ find

  93 | value={
  94 |   pluginKeys
> 95 |     .find((p) => p.pluginId === PluginID.GOOGLE_SEARCH)
     |     ^
  96 |     ?.requiredKeys.find((k) => k.key === 'GOOGLE_API_KEY')
  97 |     ?.value
  98 | }

**Comment by rpallas** [comment]
I get the same.

If I set the Google key and ID as environment variables it works fine.

If I set them in the UI, the query fails with the `pluginKeys.find is not a function` in the console. Local storage value for `pluginKeys` is set correctly but If I try to edit the key/id in the UI it also crashes with the same error. Removing `pluginKeys` from local storage allows the edit dialog to show, but of course the plugin doesn't work.
